### DISCUSSION TITLE
Marketplace tokens limited to a single extension

### DISCUSSION THREAD

**Discussion by ssbarnea** [microsoft/vscode-discussions#11] [Extension Development QnA] [answered]
At this moment you cannot generate marketplace auth tokens that give publishing rights for a single extension, they give full publishing rights for the whole publisher.

Unless you are a lonely extension developer that publishes a single extension, that counts as a serious security flaw, as it creates a big attack surface, the bigger the number of extensions a publisher has.

Leak one token from a badly configured pipeline and any other extension you published is at risk.

**Comment by isidorn** [comment]
@ssbarnea thanks for opening this discussion. I agree we need to improve here and I know there have been some changes with how tokens are generated.
@prashantvc @anangaur should know more

**Comment by prashantvc** [comment, marked as answer]
@ssbarnea appreciate your feedback! This is a know issue, we will be working on improving the PAT generation and management, this one of the goal on our specification.

**Reply by ssbarnea** [reply]
Thanks for replying, that is indeed reassuring. If you will need help testing those  upcoming changes let me know.

**Comment by robconery** [comment]
Closing for now.
